public/docs/ts/latest/cookbook/production-deploy.jade

```diff
@@ -0,0 +1,316 @@
+  [More on this later](#angular-configuration).
+  3. Configure your server to redirect requests for missing files to `index.html` instead.
+  [More on this later](#server-configuration).
+
+  And this is all you need to publish your app!
+
+  It's not a very good production deploy though. 
+  There's more you can do to make our site fast for users.
+ 
+.l-main-section
+:marked
+  ## Minimizing your payload
+
+  One of the easiest optimizations you can do is to reduce the code users have to download.
+  
+  There are several techniques for achieving a smaller payload for your app:
+
+  - Bundling: concatenating several modules into a single file (bundle). 
+  - Inlining: add html and css as strings inside components.
+  - Minification: removing all unnecessary whitespace and optional tokens.
+  - Uglification: rewriting code to use shorter variable/function names.
+  - Dead Code Elimination: removing code/variables that will never used.
+  - Tree shaking: removing unused module exports.
+  - Ahead-of-Time (AoT) Compilation: pre-compiling Angular templates.
+
+  Each of these do different things overall, but they all work together and their effects
+  compound on each other.
+
+  ### Less files
+  Bundling and inlining reduce your app's load time smaller by serving less files and thus
+  less round trips. 
+  For small files, the overwhelming majority of time it takes to get it is not spent downloading,
+  but rather communicating with the server and coordinating the transfer. 
+  You can read more about resource timings in the 
+  [Chrome DevTools Network Performance page](https://developers.google.com/web/tools/chrome-devtools/network-performance/understanding-resource-timing)   
+
+  ### Smaller files
+  Minification, Uglification and Dead Code Elimination all contribute to reduce the total size
+  of your code to the smallest possible amount of bytes, which will result in a smaller
+  transfer time.
+
+  Tree shaking also reduces the total size but does so by removing whole exports from modules.
+  It was popularized by [Rollup](http://rollupjs.org/) and you can read more about it in 
+  [this post](https://medium.com/@Rich_Harris/tree-shaking-versus-dead-code-elimination-d3765df85c80#.15ih9cyvl).
+
+  ### Template compilation
+  AoT compilation reduces render time by pre-compiling Angular templates that otherwise would be
+  compiled at runtime. 
+  It also reduces payload size since the Angular compiler itself is no longer needed, and nor are
+  directives that are never used in templates. This allows Tree-Shaking to further remove exports.
+
+  You can read more about AoT Compilation in our [dedicated cookbook chapter](aot-compiler.html),
+  where you'll also find instructions on how to use Rollup to do all the optimizations shown here.
+  Another great choice is [Webpack 2](https://webpack.js.org/) together with the 
+  [Angular Ahead-of-Time Webpack Plugin](https://github.com/angular/angular-cli/tree/master/packages/%40ngtools/webpack).
```

Comment:
Consider adding another sentence here on why this is a great choice. "Webpack makes it easier ..." or "Webpack combines ..." or whatever.